chainstate/broadcast_providers.go

```diff
@@ -94,14 +94,19 @@ func (provider broadcastClientProvider) broadcast(ctx context.Context, c *Client
 	return broadcastWithBroadcastClient(ctx, c, provider.txID, provider.txHex)
 }
 
-func broadcastWithBroadcastClient(ctx context.Context, client ClientInterface, txID, hex string) error {
+func broadcastWithBroadcastClient(ctx context.Context, client *Client, txID, hex string) error {
 	debugLog(client, txID, "executing broadcast request for "+ProviderBroadcastClient)
 
 	tx := broadcast.Transaction{
 		Hex: hex,
 	}
 
-	result, err := client.BroadcastClient().SubmitTransaction(ctx, &tx, broadcast.WithRawFormat())
+	result, err := client.BroadcastClient().SubmitTransaction(
+		ctx,
+		&tx,
+		broadcast.WithRawFormat(),
+		broadcast.WithCallback(client.options.config.callbackURL, client.options.config.callbackToken),
```

Comment:
Are `callbackURL` and `callbackToken` always properly configured? 

I'm thinking about, adding `WithCallback` when some validations of these parameters pass.
Or, storing these parameters in separated *struct; and check if it's nil.

Reply:
I was thinking about it, callback token will be always set properly (as we get the default value from xpub), callbackURL needs to be set manually.
My first thought was to add "enabled" to paymail config section, but as we follow minimal configuration approach I am not sure if it's a good idea.

The worst case scenario is that we will have localhost:3003 in callbackURL with default token, so I am not sure if I can check for nil struct or if the host is different than localhost.

Reply:
Agreed 👍